### PR TITLE
Use SecretHub to get the testacc secrets

### PR DESCRIPTION
**Pull request by florisvdg**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +15,14 @@ jobs:
           key: go-modules-{{ checksum "go.mod" }}
           paths:
             - /go/pkg/mod
-      - run: make testacc
+      - secrethub/exec:
+          step-name: Run acceptance tests
+          command: make testacc
+          # SecretHub credential to use in the acceptance tests
+          flags: -e SECRETHUB_CREDENTIAL=secrethub/terraform-provider/testacc/secrethub/credential
```
Comment (mackenbach) [inline .circleci/config.yml:22]: Just flying by here, how does this work pointing to a path instead of the contents?

Comment (florisvdg) [reply .circleci/config.yml]: `-e` just takes SecretHub paths, no `secrethub://` needed. 

To avoid potential confusion, https://github.com/secrethub/terraform-provider-secrethub/pull/83/commits/14ca082cae84511b1b1fba58d0aa82d9220b689b refactors this part to ditch the `-e` flag.

Comment (mackenbach) [reply .circleci/config.yml]: Huh so are you telling me you're getting a credential from secrethub to then call secrethub?

Comment (florisvdg) [reply .circleci/config.yml]: Indeed, that second credential is a different one than the first credential.

The first one is a simple service account that represents CircleCI setting everything up for the job and just needs read access like most CIs do.

The second one is a bit different: it's a service account with admin rights to control the _acceptance test sandbox_, which is a different workspace than the one that contains the static configuration secrets. So that second does lots of creating and destroying of resources every time the tests are triggered, because it needs to test everything the Terraform provider supports (i.e. secrets, AWS and standard service accounts, dirs, access rules) and has a very cluttered and meaningless audit log.

So long story short: 
- it's not stuff regular users would do, but that's because deploying/developing SecretHub with SecretHub is a thing only we do
- it needs to be documented somewhere

Comment (florisvdg) [reply .circleci/config.yml]: And yes: this is probably something to brainstorm about (in person).

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
 version: 2.1
+orbs:
+  secrethub: secrethub/cli@1.0.0
 jobs:
   build:
     docker:
       - image: circleci/golang:1.13
     steps:
+      - secrethub/install
```
Comment (SimonBarendse) [inline .circleci/config.yml:9]: Do you think we should add a comment explaining why we had to add this in?

File: .circleci/config.yml (hunk 1 above)
Comment (SimonBarendse) [reply .circleci/config.yml]: > it needs to be documented somewhere

Very minimal, but there's: https://github.com/secrethub/terraform-provider-secrethub#testing

File: .circleci/config.yml (hunk 2 above)
Comment (florisvdg) [reply .circleci/config.yml:9]: Not necessary IMO.
